test/quantization/fx/test_quantize_pt2e.py

```diff
@@ -299,6 +314,61 @@ def forward(self, x):
         self.checkGraphModuleNodes(
             m, expected_node_list=node_list, expected_node_occurrence=node_occurrence)
 
+    def test_prepare_qat_conv_bn_fusion(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv = torch.nn.Conv2d(3, 3, 3)
+                self.bn = torch.nn.BatchNorm2d(3)
+
+            def forward(self, x):
+                x = self.conv(x)
+                x = self.bn(x)
+                return x
+
+        import torch.ao.quantization._pt2e.quantizer.qnnpack_quantizer as qq
+        quantizer = QNNPackQuantizer()
+        quantizer.set_global(qq.get_default_per_channel_symmetric_qnnpack_operator_spec(is_qat=True))
+        m = M()
+        example_inputs = (torch.randn(1, 3, 5, 5),)
+
+        # program capture
+        m, guards = torchdynamo.export(
+            m,
+            *copy.deepcopy(example_inputs),
+            aten_graph=True,
+            tracing_mode="real",
+        )
+
+        m = prepare_qat_pt2e_quantizer(m, quantizer)
+        m(*example_inputs)
+
+        # Verify that the subgraph was replaced successfully with fake quantizes inserted
+        # Do this by matching against the fused QAT pattern with fake quantizes manually inserted
+        check_pattern = _get_aten_graph_module(
+            _fused_qat_conv_bn_pattern,
+            _conv_bn_pattern_example_inputs,
+        )
+        obs0 = copy.deepcopy(m.activation_post_process_0)  # conv input activation
+        obs1 = copy.deepcopy(m.activation_post_process_1)  # conv weight
+        obs2 = copy.deepcopy(m.activation_post_process_2)  # bn - getitem output activation
+        for node in check_pattern.graph.nodes:
+            named_modules = dict(check_pattern.named_modules(remove_duplicate=False))
+            graph = check_pattern.graph
+            if node.target == torch.ops.aten.convolution.default:
+                obs0_node = _insert_observer(node.args[0], obs0, check_pattern, named_modules, graph)
+                obs1_node = _insert_observer(node.args[1], obs1, check_pattern, named_modules, graph)
+                node.replace_input_with(node.args[0], obs0_node)
+                node.replace_input_with(node.args[1], obs1_node)
+            if node.target == operator.getitem:
+                output_node = list(node.users.keys())[0]
+                obs2_node = _insert_observer(node, obs2, check_pattern, named_modules, graph)
+                output_node.replace_input_with(node, obs2_node)
+        check_pattern.graph.eliminate_dead_code()
+        check_pattern.recompile()
+        matcher = SubgraphMatcher(check_pattern.graph)
+        self.assertEqual(len(matcher.match(m.graph)), 1)
```

Comment:
Also can you add a check for preserving metadata? or on in the follow up PR?

Reply:
Added a TODO